### PR TITLE
SONAR-22835 make securitycontext compatible with all flavor of kubernetes

### PR DESCRIPTION
**Pull request by jCOTINEAU**

### CODE REVIEW

File: charts/sonarqube-dce/templates/init-fs.yaml [hunk 1]
```diff
@@ -13,7 +13,6 @@ data:
     chown -R {{ .Values.searchNodes.persistence.uid }}:{{ .Values.searchNodes.persistence.guid }} {{ .Values.sonarqubeFolder }}/data
     chown -R {{ .Values.searchNodes.persistence.uid }}:{{ .Values.searchNodes.persistence.guid }} {{ .Values.sonarqubeFolder }}/temp
     chown -R {{ .Values.searchNodes.persistence.uid }}:{{ .Values.searchNodes.persistence.guid }} {{ .Values.sonarqubeFolder }}/logs
-    chown -R {{ .Values.searchNodes.persistence.uid }}:{{ .Values.searchNodes.persistence.guid }} {{ .Values.sonarqubeFolder }}/extensions
```
Comment (jCOTINEAU) [inline charts/sonarqube-dce/templates/init-fs.yaml:16]: not needed as extensions is there only on app nodes and persistency is on search nodes for dce

File: charts/sonarqube-dce/CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,8 @@ All changes to this chart will be documented in this file.
 * Enable the setup of ReadOnlyRootFilesystem in the security contexts
 * Support basic chart installation on Openshift
 * Support Openshift Route
+* Support Openshift SCCv2 by default when Openshift.enabled=true
+* Deprecate Openshift.createSCC
```
Comment (carminevassallo) [inline charts/sonarqube-dce/CHANGELOG.md:13]: Given that `Openshift.createSCC` gets deprecated, why don't we set the default value to false?

**Review by carminevassallo — CHANGES_REQUESTED** [review]
I've functionally validated the changes. Everything works as expected! 

There are a few suggestions on the changes that we might want to consider to further improve their usage.

File: charts/sonarqube-dce/CHANGELOG.md (hunk 2 above)
Comment (jCOTINEAU) [reply charts/sonarqube-dce/CHANGELOG.md:13]: i would prefer not to change the default to keep the compatibility and not change the behavior for the few users that might be relying on this feature.

File: charts/sonarqube/README.md [hunk 3]
```diff
@@ -203,6 +203,10 @@ If a Prometheus Operator is deployed in your cluster, you can enable a PodMonito
 
 The chart can be installed on OpenShift by setting `OpenShift.enabled=true`. Among the others, please note that this value will disable the initContainer that performs the settings required by Elasticsearch (see [here](#elasticsearch-prerequisites)). Furthermore, we strongly recommend following the [Production Use Case guidelines](#production-use-case).
 
+`Openshift.createSCC` is deprecated and should be set to `false`, the default securityContext plus the production config described [above](#production-use-case) are compatible with restricted SCCv2.
+
+If you want to try out this chart on Openshift, and use our default embedded postgresql chart please set `postgresql.securityContext.enabled=true` and `postgresql.containerSecurityContext.enabled=true`
+
```
[the remark this answers is not in the document]
Comment (jCOTINEAU) [reply charts/sonarqube/README.md:227]: did some change, let me know wyt?

**Review by carminevassallo — APPROVED** [review]
Ready to bake! LGTM!

**Pull request merged**